src/nnet3/nnet-test-utils.cc

```diff
@@ -888,13 +888,21 @@ void ComputeExampleComputationRequestSimple(
   request->outputs.clear();
   inputs->clear();
 
+  int32 t_modulus = GetTimePeriodForIvector(nnet);
   std::vector<Index> input_indexes, ivector_indexes, output_indexes;
   for (int32 n = n_offset; n < n_offset + num_examples; n++) {
     for (int32 t = input_start_frame; t < input_end_frame; t++)
       input_indexes.push_back(Index(n, t, 0));
     for (int32 t = output_start_frame; t < output_end_frame; t++)
       output_indexes.push_back(Index(n, t, 0));
-    ivector_indexes.push_back(Index(n, 0, 0));
+    if (t_modulus == 0) // case 1: single ivector for the entire chunk
```

Comment:
vijay, I think the way he has done it here might be OK... because in decoding we process multiple chunks; we are given a sequence of iVectors, and in the networks that only take iVector on time t=0, we just give it an iVector from within the chunk.  So it's valid to have t_modulus==0 [although the variable is badly named, should be ivector_interval], and still have multiple iVectors as input.  This is mostly of use for legacy networks.